src/main/java/com/google/sps/servlets/DisplayRepliedPostsServlet.java

```diff
@@ -0,0 +1,51 @@
+package com.google.sps.servlets;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for displaying posts. */
+@WebServlet("/display-posts")
+public class DisplayRepliedPostsServlet extends HttpServlet {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        // Initialize Datastore 
+        Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+
+        Query<Entity> query = Query.newEntityQueryBuilder().setKind("RepliedPost").build();
+        QueryResults<Entity> results = datastore.run(query);
+
+        // Add the post content string into the 'posts' arraylist 
+        List<List<String>> posts = new ArrayList<>();
+        while (results.hasNext()) {
+            List<String> onePost = new ArrayList<>(); 
```

Comment:
Instead of having a list of strings for a post, we can create a class for the data and convert it to gson directly, some examples: https://www.geeksforgeeks.org/convert-java-object-to-json-string-using-gson/